src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -668,6 +667,16 @@ def test_process_declarations(f2008_parser):
     assert isinstance(fake_parent.symbol_table.lookup("i5").constant_value,
                       BinaryOperation)
 
+    # Initialisation with constant expresions with symbols
+    reader = FortranStringReader("integer, parameter :: val1 = 1, val2 = val1")
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    assert fake_parent.symbol_table.lookup("val1").constant_value.value == "1"
+    assert isinstance(
+        fake_parent.symbol_table.lookup("val2").constant_value, Reference)
+    assert fake_parent.symbol_table.lookup("val2").constant_value.symbol == \
+        fake_parent.symbol_table.lookup("val1")
+
```

Comment:
Please add an example with a more complex expression e.g. `2*a-b` (to show that we can deal with this). Also add a literal with precision specified `=0.0_idef` (to show that is type of code now works).

Reply:
Added a more complex example.